### PR TITLE
week3 - lesson plan

### PR DESCRIPTION
**Pull request by gajduk**

### CODE REVIEW

**Comment by gajduk** [comment]
@NoerGitKat  do not merge WORK IN PROGRESS

**Comment by gajduk** [comment]
@NoerGitKat  can you merge this so that I can go on with my life...

File: Week3/LESSONPLAN.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+See examples from `readme`
+https://github.com/HackYourFuture/databases/blob/master/Week3/README.md#1st-normal-form-1nf-4-rules
+https://github.com/HackYourFuture/databases/blob/master/Week3/README.md#2nd-normal-form-2nf
+### Exercise
+
+Students should split in groups and discuss how to normalize the following table ![table](https://www.guru99.com/images/NormalizationTable1.png)
+5 minutes
+Ask a group of students to present their idea.
+### Essence
+What is normalization, why we do it, and the existence of normal forms. How to normalize a table to 2nd NF.
```
Comment (unmeshvrije) [inline Week3/LESSONPLAN.md]: IMHO, the essence should contain the  "why" of the concept (here Normal forms) and not questions (unless it is something that is supposed to be answered by the teacher)

File: Week3/LESSONPLAN.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+### Example
+![transaction example](https://lh3.googleusercontent.com/proxy/8RnA6wqcQxTQKJvmE7KNnhwWkWTYLXRS4lnwwpdwECLDbRoN4yEgaclhZ1pDzLi0eEHULy-_V5Lk2HkH_yFCNUa5i165WVhCR3-DvprDYem7Ai_jrP0XDSo9-L61MM4)
+
+### Exercise
+
+No time. There will be an exercise in the homework.
+
+### Essence
+THe students should now about the existence of transaction and have some idea of what they do and how to use them.
```
Comment (unmeshvrije) [inline Week3/LESSONPLAN.md]: Nitpick: THe => The and  
now => know

File: Week3/LESSONPLAN.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+### Explanation
+SQL injection is a form of hacker attack where the attacker tries to get our program to execute a query to read/write data that they should not have access to.
+
+### Example
+See examples from `readme`
+
+For use of prepared statements some minimal codes are available here: https://evertpot.com/executing-a-mysql-query-in-nodejs/
+
+### Exercise
+https://www.hacksplaining.com/exercises/sql-injection#/start
```
Comment (unmeshvrije) [inline Week3/LESSONPLAN.md]: I would rather have the exercise written down here. Also make sure that the exercise is complied with the tables and databases created in Week 1 and Week 2

File: Week3/LESSONPLAN.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+https://github.com/HackYourFuture/databases/blob/master/Week3/sql-injection.js
+
+### Essence
+Students know about the dangers of SQL injection and how to protect their app by using prepared statements.
+
+## 4. No SQL
+### Explanation
```
Comment (unmeshvrije) [inline Week3/LESSONPLAN.md]: I would use the same tables (as Week 1 lessonplan) here to be consistent and show students how to make similar databases using MySQL and NoSQL

Comment (gajduk) [reply Week3/LESSONPLAN.md]: Agreed

File: Week3/LESSONPLAN.md (hunk 3 above)
Comment (gajduk) [reply Week3/LESSONPLAN.md]: It is an interactive exercise, really cool. I will write that down

File: Week3/LESSONPLAN.md (hunk 2 above)
Comment (gajduk) [reply Week3/LESSONPLAN.md]: Fixed, I added grammarly to visual studio code, should help me make less of these silly mistakes.

File: Week3/LESSONPLAN.md (hunk 1 above)
Comment (gajduk) [reply Week3/LESSONPLAN.md]: The way I understand, essence is what the students need to know at the end of the part. Probably question is not the best format though.

**Pull request merged**